src/DetailsView/components/failure-instance-panel-details.tsx

```diff
@@ -1,38 +1,61 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
 import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import * as React from 'react';
 import { NamedSFC } from '../../common/react/named-sfc';
 
 export type FailureInstancePanelDetailsProps = {
     path: string;
-    snippet: string;
+    snippetValue: string;
+    snippetError: boolean;
     onSelectorChange: (event, value) => void;
     onValidateSelector: (event) => void;
 };
 
 export const FailureInstancePanelDetails = NamedSFC<FailureInstancePanelDetailsProps>('FailureInstancePanelDetails', props => {
+    const getSnippetInfo = (): JSX.Element => {
```

Comment:
You can improve this in terms of readability a bit by structuring it differently:

```
if (props.snippetError) {
    // return error snippet here
}

if (!props.snippetValue) {
    // return default auto populate message here

}

return 'whatever the default component is here';
```

Reply:
This can also be it's own component; should simplify it but not necessary in this PR 😄